test/directives/vector_vectorlength_fixed.f90

```diff
@@ -0,0 +1,62 @@
+! Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+! See https://llvm.org/LICENSE.txt for license information.
+! SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+! Check that the vector vectorlength(fixed) directive generates the correct metadata.
+! RUN: %flang -O0 -S -emit-llvm %s -o - | FileCheck %s --check-prefixes=CHECK-00,CHECK-ALL
+
+! Check that LLVM vectorizes the loop automatically at -O2.
+! RUN: %flang -O2 -S -emit-llvm %s -o - | FileCheck %s -check-prefixes=CHECK-O2,CHECK-ALL
+
+! Check that "-Hx,59,2" disables vector directive.
+! RUN: %flang -Hx,59,2 -S -emit-llvm %s -o - | FileCheck %s --check-prefixes=CHECK-DISABLED,CHECK-ALL
+
+subroutine func1(a, b, m)
+! CHECK-ALL: define void @func1
+  integer :: i, m, a(m), b(m)
+
+  !dir$ vector vectorlength(fixed)
+  do i = 1, m
+    b(i) = a(i) + 1
+  end do
+! CHECK-00:      [[LOOP:L.LB[0-9]_[0-9]+]]:{{[' ',\t]+}}; preds = %[[LOOP]], %L.LB
+! CHECK-00:      store i32
```

Comment:
Seems unnecessary to check this given you have identified the loop from the backedge.